platform/mime/core/platform-mime-core-impl/src/main/java/ddf/mime/mapper/MimeTypeMapperImpl.java

```diff
@@ -175,6 +179,26 @@ public String guessMimeType(InputStream is, String fileExtension)
         // from being used when a CustomMimeTypeResolver may be more appropriate.
         List<MimeTypeResolver> sortedResolvers = sortResolvers(mimeTypeResolvers);
 
+        if (StringUtils.isEmpty(fileExtension)) {
+            try (TemporaryFileBackedOutputStream tfbos = new TemporaryFileBackedOutputStream()) {
```

Comment:
Why a new filestream? Couldn't we read from the InputStream and then reset it when we are done with it? Like this?
```
if(StringUtils.isEmpty(fileExtension))) {
           try {
                Detector detector = new DefaultDetector();
                MediaType mediaType = detector.detect(is, new Metadata());
                fileExtension = getFileExtensionForMimeType(mediaType.toString()).replace(".", "");
            } catch (Exception e) {
                LOGGER.debug("Failed to guess mimeType for file without extension.");
            } finally {
                try {
                    is.reset();
                } catch (IOException e) {
                    LOGGER.debug("Failed to reset input stream.");
                }
            }
}
```